src/map.cpp

```diff
@@ -461,15 +462,52 @@ std::vector<const Enemy*> Map::find_enemies_within_circle(float center_x, float
 	for (LocationAndSection las : find_sections_within_circle(*this->priv, center_x, center_z, radius)) {
 		for (int i = 0; i < las.section->enemies.size(); i++) {
 			Enemy* enemy = &las.section->enemies.data()[i];
-			float dx = center_x - enemy->get_location().x;
-			float dz = center_z - enemy->get_location().z;
+			float dx = center_x - enemy->physics_object.location.x;
+			float dz = center_z - enemy->physics_object.location.z;
 			if (dx*dx + dz*dz < radius*radius)
 				result.push_back(&las.section->enemies.data()[i]);
 		}
 	}
 	return result;
 }
 
+std::vector<const Enemy*> Map::find_colliding_enemies(const PhysicsObject& collide_with)
+{
+	std::vector<const Enemy*> result = {};
+
+	// TODO: hard-coded 10 also appears in a few other places
+	for (const Enemy* enemy : this->find_enemies_within_circle(collide_with.location.x, collide_with.location.z, 10)) {
+		if (physics_objects_collide(collide_with, enemy->physics_object, *this))
+			result.push_back(enemy);
+	}
+	return result;
+}
+
+void Map::remove_enemies(const std::vector<const Enemy *> enemies)
+{
+	if (enemies.size() != 0)
+		log_printf("Removing %zu enemies", enemies.size());
+
+	std::unordered_map<Section*, std::vector<int>> indexes_to_delete_by_section;
+	for (const Enemy* e : enemies) {
+		Section* section = this->priv->sections[std::make_pair(get_section_start_coordinate(e->physics_object.location.x), get_section_start_coordinate(e->physics_object.location.z))].get();
+		indexes_to_delete_by_section[section].push_back(e - &section->enemies[0]);
```

Comment:
I believe iterators would behave enough like pointers, except that they give a reasonable error if they ever get invalidated?

Reply:
nope, they don't throw an exception when dereferenced